pkg/authenticator/authenticator.go

```diff
@@ -0,0 +1,19 @@
+package authenticator
+
+// DockerAuth Structure for the authentication file
+type DockerAuth struct {
+	Auths map[string]DockerAuthRegistry `json:"auths,omitempty"`
+}
+
+type DockerAuthRegistry struct {
+	Auth string `json:"auth"`
+}
+
+// Authenticator is an interface for creating an authentication file with credentials to private registries
+//
+// Currently this is used with the Helm Driver which takes credentials in this way
+// For this first implementation, kubernetes secrets will be used to pass in a token
+type Authenticator interface {
+	// GetAuthFileName Gets Authentication File Path for OCI Registry
+	GetAuthFileName() (string, error)
```

Comment:
```suggestion
	AuthFilename() (string, error)
```
Nit: There's no need for the `Get` prefix, and in go, it's usually capitalized as `Filename` not `FileName`.